python/tskit/trees.py

```diff
@@ -6351,3 +6351,124 @@ def newick_trees(self, precision=3, breakpoints=None, Ne=1):
             "This method is no longer supported. Please use the Tree.newick"
             " method instead"
         )
+
+
+def write_ms(
+    tree_sequence,
+    output,
+    print_trees=False,
+    precision=4,
+    num_replicates=1,
+    write_header=True,
+):
+    """
+    Write ``ms`` formatted output from the genotypes of a tree sequence
+    or an iterator over tree sequences. Usage:
+
+    .. code-block:: python
+
+        import tskit as ts
+
+        tree_sequence = msprime.simulate(
+            sample_size=sample_size,
+            Ne=Ne,
+            length=length,
+            mutation_rate=mutation_rate,
+            recombination_rate=recombination_rate,
+            random_seed=random_seed,
+            num_replicates=num_replicates,
+        )
+        with open("output.ms", "w") as ms_file:
+            ts.write_ms(tree_sequence, ms_file)
+
+    :param ts tree_sequence: The tree sequence (or iterator over tree sequences) to
+        write to ms file
+    :param io.IOBase output: The file-like object to write the ms-style output
+    :param bool print_trees: Boolean parameter to write out newick format trees
+        to output [optional]
+    :param int precision: Numerical precision with which to write the ms
+        output [optional]
+    :param bool write_header: Boolean parameter to write out the header. [optional]
+    :param int num_replicates: Number of replicates simulated [required if
+        num_replicates used in simulation]
+
+    The first line of this ms-style output file written has two arguments which
+    are sample size and number of replicates. The second line has a 0 as a substitute
+    for the random seed.
+    """
+    if isinstance(tree_sequence, collections.Iterable):
```

Comment:
Here you could do `tree_sequence=[tree_sequence]` if the argument wasn't an iterable. This would let you inline `print_ms_file`.

Reply:
Done this modification.